pkg/allocator/allocator.go

```diff
@@ -579,7 +594,7 @@ func (a *Allocator) lockedAllocate(ctx context.Context, key AllocatorKey) (idpoo
 		return 0, false, false, fmt.Errorf("Found master key after proceeding with new allocation for %s", k)
 	}
 
-	err = a.backend.AllocateIDIfLocked(ctx, id, key, lock)
+	key, err = a.backend.AllocateIDIfLocked(ctx, id, key, lock)
```

Comment:
Sorry, I missed this during my last review. When `AllocateIDIfLocked` returns an error, the current implementation also returns a nil key. Which means that the error message below will be incorrect.